File: 2019_AoC/Day1/star2.py
sum = 0

f = open("input.txt", "r")
for x in f:
    x2 = int(x)
    x2 = x2//3
    x2 -= 2
    sum += x2
    while x2 >0:
        x2 = x2 // 3
        x2 -= 2
        if x2 > 0:
            sum += x2

print(sum)

File: 2020_AoC/prolog/Day1/star2.pl
read_file_lines(Stream, []) :- at_end_of_stream(Stream).
read_file_lines(Stream, [X | L]) :-
  \+ at_end_of_stream(Stream),
  read_line_to_string(Stream, Y),
  number_string(X, Y),
  read_file_lines(Stream, L).

read_file(Lines) :-
  open('input.txt', read, Str),
  read_file_lines(Str, Lines),
  close(Str).

star2(L, R) :-
    member(X, L),
    member(Y, L),
    member(Z, L),
    2020 is X + Y + Z, 
    R is X * Y * Z.

main :-
  read_file(L),
  star2(L, R),
  print(R), nl.
